plugins/org.eclipse.emfcloud.modelserver.glsp.integration/src/org/eclipse/emfcloud/modelserver/glsp/EMSGLSPServer.java

```diff
@@ -54,11 +55,21 @@ protected CompletableFuture<InitializeResult> handleIntializeArgs(final Initiali
 
       try {
          ModelServerClientV2 client = createModelServerClient(modelServerURL);
-         boolean alive = client.ping().thenApply(Response<Boolean>::body).get();
-         if (alive) {
-            modelServerClientProvider.setModelServerClient(client);
+         boolean alive = false;
+         int numberOfTries = NUMBER_OF_TRIES;
+         while (!alive && numberOfTries > 0) {
+            alive = client.ping().thenApply(Response<Boolean>::body).get();
+            if (alive) {
+               modelServerClientProvider.setModelServerClient(client);
+               break;
+            }
+            numberOfTries--;
+            LOGGER.warn("Ping of modelserver did not succeed! Retrying ...");
+            Thread.sleep(1000);
```

Comment:
Perhaps we could extract the timeout also into a static variable if any adopters would like to only adapt the value?